src/renderer/angular/classes/wallet.js

```diff
@@ -251,6 +254,17 @@ class Wallet {
         promise.then((nonce) => {
             //wallet.nonceHex
 
+            if (nonce <= this.previousTransactionCount) {
+                this.generateRawTransaction(toAddressHex, valueWei, gasPriceWei, gasLimitWei, contractDataHex, chainID).then(res => {
```

Comment:
Is there some reason we're not just using a real promise here @rodrigopavezi?

Seems like we could just `return` on this line.

Reply:
@dfcowell I figured now a better way. I will need to pass the defer to the recursive function call. Then I can just return it here. Cheers